### PR TITLE
update: the new android layout, android flavors and tests

### PR DESCRIPTION
**Pull request by danil-pavlov**

### CODE REVIEW

File: docs/topics/whatsnew18.md [hunk 1]
```diff
@@ -223,6 +223,53 @@ Here are some of the key differences between the two layouts:
...
 
+#### The relation between Android tests and common tests
+
+The new Android source set layout changes the relation between Android-instrumented tests (renamed to `androidInstrumentedTest` in the new layout)
+and common tests.
+
+Previously, there was a default `dependsOn` relation between `androidAndroidTest` and `commonTest`. In practice, it meant the following:
+
+* The code in `commonTest` was available in `androidAndroidTest`.
+* `expect` declarations in `commonTest` had to have corresponding `actual` realizations in `androidAndroidTest`.
```
Comment (dsavvinov) [inline docs/topics/whatsnew18.md:234]: I think "implementation" is the correct translation for Russian "реализация", I don't think I've ever seen "realization" used in this sense in english

File: docs/topics/whatsnew18.md [hunk 2]
```diff
@@ -223,6 +223,53 @@ Here are some of the key differences between the two layouts:
...
+* Tests declared in `commonTest` were running as Android instrumented tests.
```
Comment (dsavvinov) [inline docs/topics/whatsnew18.md:235]: Minor: It reads to me personally as if they were running _only_ as Android instrumented tests, which is not correct. Maybe a rephrasing would help?

`Tests declared in commonTest were participating in Android instrumented test runs`

`Tests declared in commonTest were also running as Android instrumented tests`

`During Android instrumented tests run, all tests declared in commonTest were launched as well`

File: docs/topics/whatsnew18.md [hunk 3]
```diff
@@ -223,6 +223,53 @@ Here are some of the key differences between the two layouts:
...
+        val androidInstrumentedTest by getting {
+            dependsOn(commonTest)
+        }
+    }
+}
+```
+
+#### Support for Android flavors
+
+The Android Gradle plugin creates a handful of source sets based on `buildType`. Along with the default `debug` and `release`,
```
Comment (dsavvinov) [inline docs/topics/whatsnew18.md:254]: I think the people for whom this part is relevant are well-familiar with Android source sets, so we might cut the introduction a bit:

> Previously, source sets that corresponded to Android Source Sets with `BuildType` (`debug`/`release`) or custom flavors (e.g. `demo`/`full`) were created eagerly by Kotlin Gradle Plugin and thus accessible by constructions liks `val androidDebug by getting { ... }`
> Now they are created in the `afterEvaluate` phase. It makes such expressions ...

File: docs/topics/whatsnew18.md [hunk 4]
```diff
@@ -39,48 +39,48 @@ The new language version also includes:
 
 ### Ability to not generate TYPE_USE and TYPE_PARAMETER annotation targets
 
-If a Kotlin annotation has `TYPE` among its Kotlin targets, the annotation maps to `java.lang.annotation.ElementType.TYPE_USE` 
-in its list of Java annotation targets. This is just like how the `TYPE_PARAMETER` Kotlin target maps to 
-the `java.lang.annotation.ElementType.TYPE_PARAMETER` Java target. This is an issue for Android clients with API levels 
+If a Kotlin annotation has `TYPE` among its Kotlin targets, the annotation maps to `java.lang.annotation.ElementType.TYPE_USE`
```
Comment (dsavvinov) [inline docs/topics/whatsnew18.md:42]: Small tip for the future: I'd really recommend committing such "automatic clean-up" changes in a separate commit, as that really simplified review and, later on, reading commit history. in `kotlin.git` or `intellij.git` we always do that, and actually almost require people to commit "code reformatting" separately from the semantical changes

**Review by dsavvinov — APPROVED** [review]
Minor comments, no big notes. Thanks!

**Review by atyrin — APPROVED** [review]

File: docs/topics/whatsnew18.md (hunk 4 above)
Comment (danil-pavlov) [reply docs/topics/whatsnew18.md:42]: Sorry, I also hate to see this, but our documentation plugin did it automatically.

**Review by koshachy — APPROVED** [review]

**Pull request merged**